### PR TITLE
javascript1-week4/shahnawaz

### PR DESCRIPTION
**Pull request by shazkhan2**

### CODE REVIEW

File: 3_javascript/javascript1/week4/script4.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    return num1 * num2;
+
+} else {
+    return "Operation not supported";
+
+}
+}
+let result = calculate (3,3,"add");
+let result1 = calculate (4,12,"multiply");
+let result2 = calculate (9,3,"Nothing");
```
Comment (Elisha2605) [inline 3_javascript/javascript1/week4/script4.js:74]: Cool approach, however it doesn't follow the description of the exercise.

The exercise says: `Should be able to do simple math. fx what is 3 + 3 should respond with 6. Or what is 4 * 12 should respond with 48`.

again here the you should get the response by passing the `What is ` command followed by the operation you want to calculate. Fx: `getReply("What is 5+3")`

File: 3_javascript/javascript1/week4/script4.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+// Set a timer for 4 minutes
+setTimeout(myTimeout, 4000);
+
+function myTimeout() {
+    document.getElementById("demo").innerHTML = "I can do magic, woooooo!";
+}
```
Comment (Elisha2605) [inline 3_javascript/javascript1/week4/script4.js:81]: This should also be in the `getReply` function and will return `Timer done` when the timer is up, and the timer should get triggered by `Set a timer for [number of minutes] minutes` command.

Fx: `getReply(Set a timer for 3 minutes)`

**Review by Elisha2605 — COMMENTED** [review]
Hey Shahnawaz,

Great effort, however I noticed that you might have misunderstood the exercise requirements. Take a moment to carefully read the instructions and ask for help if needed. Don't hesitate to ask questions or seek clarification if needed. Otherwise good job!!!! Keep going and learning. You're making progress, and learning is all about taking steps forward. You've got this!

File: 3_javascript/javascript1/week4/script4.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+    return "I'm sorry, I don't understand that command.";
+}
+
+//Making a todo list and saving the values in the array aboutMe
+
+function toDoList (activity) {
+if (activity === "fishing" || activity === "singing in the shower") {
+    aboutMe.push(activity);
+return `${activity} added to your todo`;
```
[the remark this answers is not in the document]
Comment (shazkhan2) [reply 3_javascript/javascript1/week4/script4.js]: Please review again

File: 3_javascript/javascript1/week4/script4.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+function viewToDo() {
+    const excludedNames = ["Benjamin", "Ben"];
+    const activities = aboutMe.filter(activity => !excludedNames.includes(activity));
+    if (activities.length > 0) {
+        console.log(`Your todo list is: ${activities.join(", ")}`);
+    } else {
+        console.log("Your todo list is empty");
+    }
+}
```
[the remark this answers is not in the document]
Comment (shazkhan2) [reply 3_javascript/javascript1/week4/script4.js]: I had to redo it in a totally different way. please recheck

File: 3_javascript/javascript1/week4/script4.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+            return "You have already said that!";
+        }
+
+        aboutMe.push(name);
+
+        return `Nice to meet you ${name}`;
+    }
+
+    return "I'm sorry, I don't understand that command.";
+}
```
[the remark this answers is not in the document]
Comment (shazkhan2) [reply 3_javascript/javascript1/week4/script4.js]: I created if statements but struggled with passing values to it within the function. were we supposed to use the getElement.... method and input values or only function and conditions were required?

**Pull request closed without merging**